mobify/progressive-web-scaffold
# 🚀  Astro integration

This PR lands basic support for Astro integration. It forms the basis for Astro becoming a part of the Mobify Platform Scaffold. Future PRs will build on this one so that we can build/bring in the new Progressive App scaffold one piece at a time.

This PR is primarily a folder re-organization and changes to make CircleCI work. It is the base branch for https://github.com/mobify/progressive-web-scaffold/pull/236 which is the PR that forms our main "`develop`" branch until we get everything merged and working on the real `develop` branch. Once we land this branch we will re-target https://github.com/mobify/progressive-web-scaffold/pull/236 to use `develop` as the base branch.

 **JIRA**: https://mobify.atlassian.net/browse/HYB-962
 **Linked PRs**: https://github.com/mobify/progressive-web-scaffold/pull/236, https://github.com/mobify/astro/pull/744, https://github.com/mobify/astro-scaffold/pull/155

## Outstanding questions
There are several questions that we need to address while we integrate the app and web platforms into a single scaffold (ie. in this PR):
- [x] How will we configure CircleCI?
       _[@jeremywiebe] There will be very little configuration needed for progressive apps as we primarily use BuddyBuild for CI. Some of this might be dependent on our decision of how to manage branching (only do certain build actions for web/app depending on the branch)._
- [x] What will the branching strategy be like?
       _[@jeremywiebe] All branches will be taken from `develop`. PRs get labelled with `web` or `app` or `platform` depending on who needs to review the code._
- [x] Will we release everything at once or still be able to release app vs web features individually. How?
       _[@jeremywiebe] At this point the scaffold is a "continuous release" type of project (we are always generating based on whatever is on `master`). I think in the near future we should move to deprecate the `develop` branch and use release branches to host development of different teams (as we've discussed already). This will keep us out of each other's hair and prevent unreleasable code from being merged to `master`._
- [x] How will we mark web / app PRs to make sure the right people review the right PRs?
       Maybe use labels? `web` and `astro`?
       _[@jeremywiebe] Yep, labels!_

## Changes
- Trigger Astro event in app-provider's `onChange` if running in an Astro app.

## How to test-drive this PR
- Run the tests: `cd web && npm test`
- Ensure CircleCI passes
- Run the web scaffold and make sure it works
